### PR TITLE
Support Mushaf API

### PR DESCRIPTION
**Pull request by osamasayed**
### Summary
This PR adds support to change the Mushaf based on user selection of the font inside the Settings Drawer. This primarily adds support to have 15-line and 16-line IndoPak Mushafs.

### Screenshots

| Before | After |
| ------ | ------ |
|Screen Shot 2021-09-28 at 12 55 10|Screen Shot 2021-09-28 at 12 54 53|

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/\[chapterId\]/\[verseId\]**](https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/[chapterId]/[verseId]) | [**/\[chapterId\]**](https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/[chapterId]) |
|:-:|:-:|
|<a href="https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/[chapterId]/[verseId]">Screenshot of /[chapterId]/[verseId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/pQC3rI4I1_q_WIDN0YTXDU-fkupFoHF_lRLIqhoQjeLVVA7lmSLU6BWoWypWpi8BgsurZtRQgzmAMTtxYNob7w.png">(view full size)</a> |<a href="https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/[chapterId]">Screenshot of /[chapterId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/RG44AOohoeSZ8KRFoCemUABc6kFCWTFN51vt5JHLGidNHEYsrcgGyX-7F4fVwJPi5i6YhEHW8sDN57N7wvaS4Q.png">(view full size)</a> |



| [**/juz/\[juzId\]**](https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/juz/[juzId]) | [**/page/\[pageId\]**](https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/page/[pageId]) |
|:-:|:-:|
|<a href="https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/juz/[juzId]">Screenshot of /juz/[juzId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/-s2lj7tIsMdTS49uc-j-yVXXTGEqjilzcTa7J7dIRM1R0MCd3Fn-S6uRFJ6RvP1j0uEfwUnJNHkFeStKoElnMA.png">(view full size)</a> |<a href="https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app/page/[pageId]">Screenshot of /page/[pageId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/LV4phltx9fCz1P_Sc_jKCA51WKxhSsFFPcvjyJgAKkKCAl0I4pPAIgBqPMJ3ftwECT7kxfQJAtQAaAv5v6IfAw.png">(view full size)</a> |







Commit 0fa52086219faeac0bc2e760a0bfd87ef989b903 (https://quran-com-frontend-next-2jqhgrggt-qurancom.vercel.app).

File: src/redux/slices/QuranReader/styles.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export type QuranReaderStyles = {
   translationFontScale: number;
   quranTextFontScale: number;
   quranFont: QuranFont;
+  mushafLines: MushafLine;
```
Comment (tifa2UP) [inline src/redux/slices/QuranReader/styles.ts:15]: should this be MushafLines?

File: src/styles/fonts.scss [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 @font-face {
   font-family: "IndoPak";
-  src: url("/fonts/indoPak/_PDMS_Saleem_QuranFont.ttf") format("truetype");
+  src: url("/fonts/indoPak/Naskh-Nastaleeq-IndoPak.woff"),
```
Comment (tifa2UP) [inline src/styles/fonts.scss:8]: nit: add format here

```
@font-face {
  font-family: 'MyWebFont';
  src:  url('myfont.woff2') format('woff2'),
        url('myfont.woff') format('woff');
}
```

File: src/utils/page.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+
+const DEFAULT_NUMBER_OF_PAGES = 604;
 
 // a map between the mushafId and the number of pages it has
-const MUSHAF_PAGES = {
-  5: 604,
+const PAGES_MUSHAF_MAP = {
+  7: 548,
+  6: 610,
 };
```
Comment (tifa2UP) [inline src/utils/page.ts:9]: Should we add the other Ids here? 5 was previously added as a sample

File: src/components/QuranReader/types.ts [hunk 4]
```diff
@@ -28,4 +28,27 @@ export enum QuranFont {
...
+}
+
+export const QuranFontMushaf: Record<QuranFont, Mushaf> = {
+  [QuranFont.MadaniV1]: Mushaf.QCFV1,
+  [QuranFont.MadaniV2]: Mushaf.QCFV2,
+  [QuranFont.Uthmani]: Mushaf.UthmaniHafs,
+  [QuranFont.IndoPak]: Mushaf.Indopak,
+  [QuranFont.QPCHafs]: Mushaf.KFGQPCHAFS,
+};
+
```
Comment (tifa2UP) [inline src/components/QuranReader/types.ts:53]: Not related to this PR, but have a feeling that this file needs to be moved out of components to the types directory.

File: src/components/QuranReader/types.ts [hunk 5]
```diff
@@ -28,4 +28,27 @@ export enum QuranFont {
   QPCHafs = 'qpc_uthmani_hafs',
 }
 
+export enum MushafLine {
+  FifteenLines = '15_lines',
+  SixteenLines = '16_lines',
+}
```
Comment (tifa2UP) [inline src/components/QuranReader/types.ts:34]: IMO this should be MushafLines, each property of the enum is a group of lines, not a single line

File: src/components/Navbar/SettingsDrawer/QuranFontSection.tsx [hunk 6]
```diff
@@ -106,6 +122,18 @@ const QuranFontSection = () => {
           onChange={(value) => dispatch(setQuranFont(value as QuranFont))}
         />
       </Section.Row>
+      {selectedType === QuranFont.IndoPak && (
+        <Section.Row>
+          <Section.Label>Mushaf lines</Section.Label>
```
Comment (tifa2UP) [inline src/components/Navbar/SettingsDrawer/QuranFontSection.tsx:127]: Lines

**Pull request merged**